### DISCUSSION TITLE
client-ts: nextjs example Keplr login fails; error message vague; docs missing

### DISCUSSION THREAD

**Discussion by a3s7p** [NillionNetwork/discussions#138] [Bugs]
1) Naive log in via Keplr results in the button eternally "loading" and this in the JS console:

![image](https://github.com/user-attachments/assets/334bac66-8e8b-410b-8001-6bc180b6c76b)

Fixed by first running `npx @nillion/add-devnet-chain-to-keplr` and going through the process to add chain & a funded wallet private key to Keplr.

2) Same thing with slightly different stack trace actually happens if Keplr is not at all installed.

Suggestions:
- produce a more useful error message in these cases
  - expose any errors to the user in error pop-up?
- document the prerequisites better
  - in the repo readme
  - as text on the actual example webpage
- integrate `@nillion/add-devnet-chain-to-keplr` into the example?
  - add more documentation there as well
  - enable/document using it from other code — perhaps package as reusable next.js component?

**Comment by crypblizz8** [comment]
Looking into this today - thanks @a3s7p
